agent/main.go

```diff
@@ -109,12 +110,20 @@ func main() {
 	sharedInformers := informers.NewSharedInformerFactory(k8sCS, 10*time.Minute)
 
 	var l5dApi *l5dk8s.KubernetesAPI
+	var k8sDC dynamic.Interface
+
 	if *localMode {
+		// in this case we can reuse the dynamic client from Linkerd
+		// as it is already there
 		l5dApi, err = l5dk8s.NewAPIForConfig(k8sConfig, "", nil, 0)
 		dieIf(err)
+		k8sDC = l5dApi.DynamicClient
+	} else {
+		k8sDC, err = dynamic.NewForConfig(k8sConfig)
+		dieIf(err)
```

Comment:
~isn't the dynamic client also available if localMode is false?~

~also now that i'm asking this, i'm trying to remember why we don't also just use `l5dApi.Interface` rather than creating our own `k8sCS`?~

ok, now i'm remembering that we instantiate `l5dApi` strictly for local mode, to enable port-forwarding. recommend always calling `dynamic.NewForConfig(k8sConfig)` ourselves, similar to how we create `k8sCS`, and add a comment explaining why we're calling `l5dk8s.NewAPIForConfig()` for local mode.

down the road, we could consider piggy-backing more on `l5dApi`, it looks like it's doing very similar things to our k8s client.